# Profitable App Profiles for Apple & Google App Stores

The goal for this project is to analyze data to understand what kinds of apps are likely to attract more users.

Datasets sourced from:\
https://www.kaggle.com/datasets/muneebullahmuneeb/googleplaystore-dataset

https://www.kaggle.com/datasets/ysf12ff/app-store-dataset

In [252]:
import pandas as pd
import numpy as np

apple=pd.read_csv('/Users/joeytan/Documents/Data Analyst/Portfolio/PortfolioProject3_App_Stores /AppleStore.csv')
google=pd.read_csv('/Users/joeytan/Documents/Data Analyst/Portfolio/PortfolioProject3_App_Stores /googleplaystore.csv')

# Removing Duplicates

In [255]:
apple[apple.duplicated()]

Empty DataFrame
Columns: [id, track_name, size_bytes, currency, price, rating_count_tot, rating_count_ver, user_rating, user_rating_ver, ver, cont_rating, prime_genre, sup_devices.num, ipadSc_urls.num, lang.num, vpp_lic]
Index: []

In [257]:
google[google.duplicated()]

App      Category  Rating  \
229                       Quick PDF Scanner + OCR FREE      BUSINESS     4.2   
236                                                Box      BUSINESS     4.2   
239                                 Google My Business      BUSINESS     4.4   
256                                ZOOM Cloud Meetings      BUSINESS     4.4   
261                          join.me - Simple Meetings      BUSINESS     4.0   
...                                                ...           ...     ...   
8643                    Wunderlist: To-Do List & Tasks  PRODUCTIVITY     4.6   
8654   TickTick: To Do List with Reminder, Day Planner  PRODUCTIVITY     4.6   
8658                           ColorNote Notepad Notes  PRODUCTIVITY     4.6   
10049        Airway Ex - Intubate. Anesthetize. Train.       MEDICAL     4.3   
10768                                             AAFP       MEDICAL     3.8   

       Reviews                Size      Installs  Type Price Content Rating  \
229      80805  Varies with device    5,000,000+  Free     0       Everyone   
236     159872  Varies with device   10,000,000+  Free     0       Everyone   
239      70991  Varies with device    5,000,000+  Free     0       Everyone   
256      31614                 37M   10,000,000+  Free     0       Everyone   
261       6989  Varies with device    1,000,000+  Free     0       Everyone   
...        ...                 ...           ...   ...   ...            ...   
8643    404610  Varies with device   10,000,000+  Free     0       Everyone   
8654     25370  Varies with device    1,000,000+  Free     0       Everyone   
8658   2401017  Varies with device  100,000,000+  Free     0       Everyone   
10049      123                 86M       10,000+  Free     0       Everyone   
10768       63                 24M       10,000+  Free     0       Everyone   

             Genres       Last Updated         Current Ver         Android Ver  
229        Business  February 26, 2018  Varies with device        4.0.3 and up  
236        Business      July 31, 2018  Varies with device  Varies with device  
239        Business      July 24, 2018    2.19.0.204537701          4.4 and up  
256        Business      July 20, 2018      4.1.28165.0716          4.0 and up  
261        Business      July 16, 2018           4.3.0.508          4.4 and up  
...             ...                ...                 ...                 ...  
8643   Productivity      April 6, 2018  Varies with device  Varies with device  
8654   Productivity     August 6, 2018  Varies with device  Varies with device  
8658   Productivity      June 27, 2018  Varies with device  Varies with device  
10049       Medical       June 1, 2018              0.6.88          5.0 and up  
10768       Medical      June 22, 2018               2.3.1          5.0 and up  

[483 rows x 13 columns]

In [259]:
google[(google['App'].str.contains('Quick PDF'))]

App  Category  Rating Reviews  \
222  Quick PDF Scanner + OCR FREE  BUSINESS     4.2   80805   
229  Quick PDF Scanner + OCR FREE  BUSINESS     4.2   80805   
285  Quick PDF Scanner + OCR FREE  BUSINESS     4.2   80804   

                   Size    Installs  Type Price Content Rating    Genres  \
222  Varies with device  5,000,000+  Free     0       Everyone  Business   
229  Varies with device  5,000,000+  Free     0       Everyone  Business   
285  Varies with device  5,000,000+  Free     0       Everyone  Business   

          Last Updated         Current Ver   Android Ver  
222  February 26, 2018  Varies with device  4.0.3 and up  
229  February 26, 2018  Varies with device  4.0.3 and up  
285  February 26, 2018  Varies with device  4.0.3 and up

If we look at the reviews column, we can see that entry 285 is not actually an exact duplicate.\
We only want to keep the latest entry of the apps, for that we have to sort by the 'last updated' and 'current ver' column

In [262]:
google.sort_values(by='Last Updated', ascending = False)

App            Category  Rating  \
6834                          Kim Bu Youtuber?              FAMILY     NaN   
10393             SnowMobile Parking Adventure                GAME     3.6   
9437                                      EI国际              FAMILY     4.7   
6983            Best Friends Dress Up & Makeup              FAMILY     3.7   
2952                         Add Text To Photo         PHOTOGRAPHY     4.1   
...                                        ...                 ...     ...   
9116                          PHARMAGUIDE (DZ)  HEALTH_AND_FITNESS     4.1   
8901              DU Browser—Browse fast & fun       COMMUNICATION     4.3   
3304                       Graphing Calculator               TOOLS     3.9   
424               DU Browser—Browse fast & fun       COMMUNICATION     4.3   
10472  Life Made WI-Fi Touchscreen Photo Frame                 1.9    19.0   

       Reviews    Size     Installs  Type     Price Content Rating  \
6834        11    9.2M       1,000+  Free         0       Everyone   
10393    12257     49M   1,000,000+  Free         0       Everyone   
9437        15    5.2M       1,000+  Free         0       Everyone   
6983      7664     44M     500,000+  Free         0       Everyone   
2952     21578    1.6M   1,000,000+  Free         0       Everyone   
...        ...     ...          ...   ...       ...            ...   
9116       131    3.4M       5,000+  Free         0       Everyone   
8901   1133539    4.7M  10,000,000+  Free         0       Everyone   
3304     12388    5.3M   1,000,000+  Free         0       Everyone   
424    1133501    4.7M  10,000,000+  Free         0       Everyone   
10472     3.0M  1,000+         Free     0  Everyone            NaN   

                    Genres       Last Updated Current Ver   Android Ver  
6834                Puzzle  September 9, 2017         1.0    4.0 and up  
10393               Racing  September 9, 2016         1.4    2.3 and up  
9437             Education  September 9, 2016       1.0.4    4.0 and up  
6983   Casual;Pretend Play  September 9, 2015       1.0.3  4.0.3 and up  
2952           Photography  September 8, 2017       1.1.3    2.3 and up  
...                    ...                ...         ...           ...  
9116      Health & Fitness      April 1, 2016        1.06    1.6 and up  
8901         Communication      April 1, 2016     6.4.0.4    4.0 and up  
3304                 Tools      April 1, 2016        1.17    2.3 and up  
424          Communication      April 1, 2016     6.4.0.4    4.0 and up  
10472    February 11, 2018             1.0.19  4.0 and up           NaN  

[10841 rows x 13 columns]

It seems that column is in a string format, we'll have to change it to date format in order to sort it

In [265]:
google['Last Updated']= pd.to_datetime(google['Last Updated'])
google.sort_values(by='Last Updated', ascending = False)

ValueError: time data "1.0.19" doesn't match format "%B %d, %Y", at position 1348. You might want to try:
    - passing `format` if your strings have a consistent format;
    - passing `format='ISO8601'` if your strings are all ISO8601 but not necessarily in exactly the same format;
    - passing `format='mixed'`, and the format will be inferred for each element individually. You might want to use `dayfirst` alongside this.

In [267]:
google[(google['Last Updated']== '1.0.19')]
#it seems this entry is inputted wrongly
google.drop(google[(google['Last Updated']== '1.0.19')].index, inplace =True)
google['Last Updated']= pd.to_datetime(google['Last Updated'])
google = google.sort_values(by=['Last Updated','Current Ver'], ascending = False)
google

App            Category  \
10712                         Lalafo Pulsuz Elanlar            SHOPPING   
10718                                    BankNordik             FINANCE   
10760                               Fast Tract Diet  HEALTH_AND_FITNESS   
10209  Video Downloader For FB: Save FB Videos 2018               TOOLS   
10408                       Shoot Hunter-Gun Killer                GAME   
...                                             ...                 ...   
8084                                    Freecell CY                GAME   
8418                          DEER HUNTER CHALLENGE                GAME   
10282                      FD Shift Calendar Widget               TOOLS   
7430                       CJ Poker Odds Calculator                GAME   
7479                      FML F*ck my life + widget              FAMILY   

       Rating Reviews                Size     Installs  Type  Price  \
10712     4.4   61392  Varies with device   1,000,000+  Free      0   
10718     3.9      28                 15M       5,000+  Free      0   
10760     4.4      35                2.4M       1,000+  Paid  $7.99   
10209     4.3     434                3.6M      50,000+  Free      0   
10408     4.3  320334                 27M  50,000,000+  Free      0   
...       ...     ...                 ...          ...   ...    ...   
8084      4.0     387                1.1M      50,000+  Free      0   
8418      3.7   38767                4.1M   5,000,000+  Free      0   
10282     4.1     981                 73k     100,000+  Free      0   
7430      4.1     207                116k      50,000+  Free      0   
7479      4.2    1415                209k     100,000+  Free      0   

      Content Rating            Genres Last Updated         Current Ver  \
10712       Everyone          Shopping   2018-08-08  Varies with device   
10718       Everyone           Finance   2018-08-08               7.3.2   
10760       Everyone  Health & Fitness   2018-08-08               1.9.3   
10209       Everyone             Tools   2018-08-08               1.1.9   
10408           Teen            Action   2018-08-08               1.1.2   
...              ...               ...          ...                 ...   
8084        Everyone              Card   2011-04-16               1.3.3   
8418    Everyone 10+            Action   2011-04-11               1.4.3   
10282       Everyone             Tools   2011-03-16               1.2.4   
7430        Everyone              Card   2011-01-30                 1.2   
7479        Everyone     Entertainment   2010-05-21                 3.1   

              Android Ver  
10712  Varies with device  
10718          5.0 and up  
10760          4.2 and up  
10209        4.0.3 and up  
10408          4.1 and up  
...                   ...  
8084           1.6 and up  
8418           2.0 and up  
10282          1.6 and up  
7430           1.6 and up  
7479           1.5 and up  

[10840 rows x 13 columns]

In [269]:
google = google.drop_duplicates(subset=['App'])
google[(google['App'].str.contains('Quick PDF'))]

App  Category  Rating Reviews  \
222  Quick PDF Scanner + OCR FREE  BUSINESS     4.2   80805   

                   Size    Installs  Type Price Content Rating    Genres  \
222  Varies with device  5,000,000+  Free     0       Everyone  Business   

    Last Updated         Current Ver   Android Ver  
222   2018-02-26  Varies with device  4.0.3 and up

# Removing Redundant and Irrelevant Data

According to the Apple data documentation

"id" : App ID\
"track_name": App Name\
"size_bytes": Size (in Bytes)\
"currency": Currency Type\
"price": Price amount\
"rating_count_tot": User Rating counts (for all version)\
"rating_count_ver": User Rating counts (for current version)\
"user_rating" : Average User Rating value (for all version)\
"user_rating_ver": Average User Rating value (for current version)\
"ver" : Latest version code\
"cont_rating": Content Rating\
"prime_genre": Primary Genre\
"sup_devices.num": Number of supporting devices\
"ipadSc_urls.num": Number of screenshots showed for display\
"lang.num": Number of supported languages\
"vpp_lic": Vpp Device Based Licensing Enabled

we can remove some of the redundant columns such as 'rating_count_ver' and 'user_rating_ver'\
we can also remove some irrelevant data that should not affect our analysis such as the app id and version number.

For google, the 'category' and 'genres' columns are quite similar, we can just keep 1 of it

In [272]:
apple.info()
google.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 7197 entries, 0 to 7196
Data columns (total 16 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   id                7197 non-null   int64  
 1   track_name        7197 non-null   object 
 2   size_bytes        7197 non-null   int64  
 3   currency          7197 non-null   object 
 4   price             7197 non-null   float64
 5   rating_count_tot  7197 non-null   int64  
 6   rating_count_ver  7197 non-null   int64  
 7   user_rating       7197 non-null   float64
 8   user_rating_ver   7197 non-null   float64
 9   ver               7197 non-null   object 
 10  cont_rating       7197 non-null   object 
 11  prime_genre       7197 non-null   object 
 12  sup_devices.num   7197 non-null   int64  
 13  ipadSc_urls.num   7197 non-null   int64  
 14  lang.num          7197 non-null   int64  
 15  vpp_lic           7197 non-null   int64  
dtypes: float64(3), int64(8), object(5)
memory 

In [274]:
apple=apple.drop(apple.columns[[0,6,8,9,13]],axis =1)
google=google.drop(google.columns[[9,10,11]],axis =1)

# Standardizing Data

we want the data to be standardised in terms of currency and category/genres

In [277]:
apple[(apple['currency']!='USD')]

Empty DataFrame
Columns: [track_name, size_bytes, currency, price, rating_count_tot, user_rating, cont_rating, prime_genre, sup_devices.num, lang.num, vpp_lic]
Index: []

There are no other currencies aside from USD in the currency column, making it redudant

In [280]:
apple=apple.drop(columns = 'currency')

In [282]:
apple['prime_genre'].unique()

array(['Social Networking', 'Photo & Video', 'Games', 'Music',
       'Reference', 'Health & Fitness', 'Weather', 'Utilities', 'Travel',
       'Shopping', 'News', 'Navigation', 'Lifestyle', 'Entertainment',
       'Food & Drink', 'Sports', 'Book', 'Finance', 'Education',
       'Productivity', 'Business', 'Catalogs', 'Medical'], dtype=object)

In [284]:
google['Category'].unique()

array(['SHOPPING', 'FINANCE', 'HEALTH_AND_FITNESS', 'TOOLS', 'GAME',
       'FAMILY', 'NEWS_AND_MAGAZINES', 'COMMUNICATION', 'VIDEO_PLAYERS',
       'PARENTING', 'LIFESTYLE', 'SPORTS', 'AUTO_AND_VEHICLES', 'WEATHER',
       'BUSINESS', 'SOCIAL', 'MAPS_AND_NAVIGATION', 'PERSONALIZATION',
       'PHOTOGRAPHY', 'TRAVEL_AND_LOCAL', 'PRODUCTIVITY',
       'FOOD_AND_DRINK', 'MEDICAL', 'BEAUTY', 'BOOKS_AND_REFERENCE',
       'HOUSE_AND_HOME', 'ENTERTAINMENT', 'DATING', 'EDUCATION',
       'ART_AND_DESIGN', 'EVENTS', 'COMICS', 'LIBRARIES_AND_DEMO'],
      dtype=object)

In [286]:
apple['cont_rating'].unique()

array(['4+', '12+', '9+', '17+'], dtype=object)

In [288]:
google['Type'].unique()

array(['Free', 'Paid', nan], dtype=object)

In [290]:
google[(google['Type'].isna())]

App Category  Rating Reviews                Size  \
9148  Command & Conquer: Rivals   FAMILY     NaN       0  Varies with device   

     Installs Type Price Content Rating         Android Ver  
9148        0  NaN     0   Everyone 10+  Varies with device

This app seems to be wrongly inputted, but i have a feeling that this column is redundant as the price column should tell us whether or not the app is paid or free.

In [293]:
google[((google['Type']=='Free')&(google['Price']!='0'))]

Empty DataFrame
Columns: [App, Category, Rating, Reviews, Size, Installs, Type, Price, Content Rating, Android Ver]
Index: []

Just as i thought, there are no free apps where the price is not 0, we can drop this column

In [296]:
google = google.drop(columns = 'Type')

I also noticed the price in the google dataset is in a string format, we won't be able to aggregate on this unless we change the datatype.\
Likewise for installs and reviews in google.

In [299]:
google['Price']=google['Price'].str.strip('$')
google['Price']=google['Price'].apply(pd.to_numeric)
google['Installs']=google['Installs'].str.strip('+')
google['Installs']=google['Installs'].str.replace(',','')
google['Installs']=google['Installs'].astype(int)
google['Reviews']=google['Reviews'].astype(int)

# EDA

# Looking at market share of each genre

In [303]:
(apple.groupby('prime_genre')['prime_genre'].count()/len(apple)*100).round(2).sort_values(ascending=False)

prime_genre
Games                53.66
Entertainment         7.43
Education             6.29
Photo & Video         4.85
Utilities             3.45
Health & Fitness      2.50
Productivity          2.47
Social Networking     2.32
Lifestyle             2.00
Music                 1.92
Shopping              1.70
Sports                1.58
Book                  1.56
Finance               1.45
Travel                1.13
News                  1.04
Weather               1.00
Reference             0.89
Food & Drink          0.88
Business              0.79
Navigation            0.64
Medical               0.32
Catalogs              0.14
Name: prime_genre, dtype: float64

In [305]:
(google.groupby('Category')['Category'].count()/len(google)*100).round(2).sort_values(ascending=False)

Category
FAMILY                 19.00
GAME                    9.93
TOOLS                   8.58
BUSINESS                4.34
MEDICAL                 4.09
PERSONALIZATION         3.89
PRODUCTIVITY            3.87
LIFESTYLE               3.82
FINANCE                 3.57
SPORTS                  3.36
COMMUNICATION           3.27
HEALTH_AND_FITNESS      2.98
PHOTOGRAPHY             2.91
NEWS_AND_MAGAZINES      2.63
SOCIAL                  2.47
BOOKS_AND_REFERENCE     2.30
TRAVEL_AND_LOCAL        2.27
SHOPPING                2.09
DATING                  1.76
VIDEO_PLAYERS           1.69
MAPS_AND_NAVIGATION     1.36
EDUCATION               1.20
FOOD_AND_DRINK          1.16
ENTERTAINMENT           1.06
AUTO_AND_VEHICLES       0.88
LIBRARIES_AND_DEMO      0.87
WEATHER                 0.82
HOUSE_AND_HOME          0.76
EVENTS                  0.66
ART_AND_DESIGN          0.66
PARENTING               0.62
COMICS                  0.58
BEAUTY                  0.55
Name: Category, dtype: float64

# Looking at average number of users within each genre / category

In [307]:
apple.groupby('prime_genre')['rating_count_tot'].mean().sort_values(ascending=False)

prime_genre
Social Networking    45498.898204
Music                28842.021739
Reference            22410.843750
Weather              22181.027778
Shopping             18615.327869
Photo & Video        14352.280802
Travel               14129.444444
Sports               14026.929825
Food & Drink         13938.619048
Games                13691.996634
News                 13015.066667
Navigation           11853.956522
Finance              11047.653846
Health & Fitness      9913.172222
Productivity          8051.325843
Entertainment         7533.678505
Utilities             6863.822581
Lifestyle             6161.763889
Book                  5125.437500
Business              4788.087719
Education             2239.229581
Catalogs              1732.500000
Medical                592.782609
Name: rating_count_tot, dtype: float64

In [309]:
google.groupby('Category')['Installs'].mean().round().sort_values(ascending=False)

Category
COMMUNICATION          34934419.0
VIDEO_PLAYERS          24091428.0
SOCIAL                 22961790.0
ENTERTAINMENT          20722157.0
PHOTOGRAPHY            16577038.0
PRODUCTIVITY           15489549.0
GAME                   14461861.0
TRAVEL_AND_LOCAL       13218663.0
TOOLS                   9774152.0
NEWS_AND_MAGAZINES      9327629.0
BOOKS_AND_REFERENCE     7504367.0
SHOPPING                6932420.0
WEATHER                 4570893.0
PERSONALIZATION         4075784.0
HEALTH_AND_FITNESS      3972300.0
MAPS_AND_NAVIGATION     3841846.0
SPORTS                  3373768.0
FAMILY                  2527489.0
EDUCATION               2090103.0
FOOD_AND_DRINK          1891060.0
ART_AND_DESIGN          1786533.0
BUSINESS                1651945.0
LIFESTYLE               1365375.0
HOUSE_AND_HOME          1331541.0
FINANCE                 1319851.0
DATING                   828971.0
COMICS                   803235.0
LIBRARIES_AND_DEMO       630904.0
AUTO_AND_VEHICLES        625061.0
PAREN

# looking at the most used apps

In [314]:
apple.sort_values(by='rating_count_tot',ascending = False).head(10)

track_name  size_bytes  price  rating_count_tot  user_rating  \
0                 Facebook   389879808    0.0           2974676          3.5   
1                Instagram   113954816    0.0           2161558          4.5   
2           Clash of Clans   116476928    0.0           2130805          4.5   
3               Temple Run    65921024    0.0           1724546          4.5   
4  Pandora - Music & Radio   130242560    0.0           1126879          4.0   
5                Pinterest    74778624    0.0           1061624          4.5   
6                    Bible    92774400    0.0            985920          4.5   
7         Candy Crush Saga   222846976    0.0            961794          4.5   
8            Spotify Music   132510720    0.0            878563          4.5   
9              Angry Birds   175966208    0.0            824451          4.5   

  cont_rating        prime_genre  sup_devices.num  lang.num  vpp_lic  
0          4+  Social Networking               37        29        1  
1         12+      Photo & Video               37        29        1  
2          9+              Games               38        18        1  
3          9+              Games               40         1        1  
4         12+              Music               37         1        1  
5         12+  Social Networking               37        27        1  
6          4+          Reference               37        45        1  
7          4+              Games               43        24        1  
8         12+              Music               37        18        1  
9          4+              Games               38        10        1

In [316]:
google.sort_values(by='Installs',ascending = False).head(10)

App             Category  Rating   Reviews  \
1654                 Subway Surfers                 GAME     4.5  27722264   
340                           Gmail        COMMUNICATION     4.3   4604324   
3454                   Google Drive         PRODUCTIVITY     4.4   2731171   
2808                  Google Photos          PHOTOGRAPHY     4.5  10858556   
865               Google Play Games        ENTERTAINMENT     4.3   7165362   
152               Google Play Books  BOOKS_AND_REFERENCE     3.9   1433233   
2554                        Google+               SOCIAL     4.2   4831125   
336              WhatsApp Messenger        COMMUNICATION     4.4  69119316   
391   Skype - free IM & video calls        COMMUNICATION     4.1  10484169   
3687        Google Play Movies & TV        VIDEO_PLAYERS     3.7    906384   

                    Size    Installs  Price Content Rating         Android Ver  
1654                 76M  1000000000    0.0   Everyone 10+          4.1 and up  
340   Varies with device  1000000000    0.0       Everyone  Varies with device  
3454  Varies with device  1000000000    0.0       Everyone  Varies with device  
2808  Varies with device  1000000000    0.0       Everyone  Varies with device  
865   Varies with device  1000000000    0.0           Teen  Varies with device  
152   Varies with device  1000000000    0.0           Teen  Varies with device  
2554  Varies with device  1000000000    0.0           Teen  Varies with device  
336   Varies with device  1000000000    0.0       Everyone  Varies with device  
391   Varies with device  1000000000    0.0       Everyone  Varies with device  
3687  Varies with device  1000000000    0.0           Teen  Varies with device

The apps above have a high number of users, influencing the average user numbers within each genre.\
To get a more accurate representation of the popularity of the apps, we must remove the outliers with way above average users
# Creating outlier mask

In [325]:
Q1 = apple['rating_count_tot'].quantile(0.25)
Q3 = apple['rating_count_tot'].quantile(0.75)
IQR = Q3 - Q1
lower = Q1 - 1.5*IQR
upper = Q3 + 1.5*IQR

upper_mask = apple['rating_count_tot'] >= upper
lower_mask = apple['rating_count_tot'] <= lower

outlier_mask = ~(upper_mask | lower_mask)
apple_mask = apple[outlier_mask]
apple_mask.sort_values(by='rating_count_tot',ascending = False).head()

track_name  size_bytes  price  \
1231  Christmas Dentist Doctor Kid Games (Girls & Boys)    38051840   0.00   
1232                                          Junk Jack   161118208   4.99   
1233                                        Battleheart   275481600   2.99   
1234                                Best Fiends Forever   210092032   0.00   
1235                                          No Brakes    21550080   0.00   

      rating_count_tot  user_rating cont_rating prime_genre  sup_devices.num  \
1231              6917          4.5          4+       Games               40   
1232              6891          4.5          4+       Games               37   
1233              6879          4.5          9+       Games               38   
1234              6872          4.5          4+       Games               37   
1235              6865          4.5          4+       Games               40   

      lang.num  vpp_lic  
1231         1        1  
1232         1        1  
1233         1        1  
1234        11        1  
1235         1        1

In [327]:
Q1 = google['Installs'].quantile(0.25)
Q3 = google['Installs'].quantile(0.75)
IQR = Q3 - Q1
lower = Q1 - 1.5*IQR
upper = Q3 + 1.5*IQR

upper_mask = google['Installs'] >= upper
lower_mask = google['Installs'] <= lower

outlier_mask = ~(upper_mask | lower_mask)
google_mask = google[outlier_mask]
google_mask.sort_values(by='Installs',ascending = False).head()

App       Category  Rating  Reviews  \
10712          Lalafo Pulsuz Elanlar       SHOPPING     4.4    61392   
6167   Block Gun 3D: Call of Destiny           GAME     3.7    31883   
3681           HD Movie Video Player  VIDEO_PLAYERS     4.5    18699   
9551     Learn to code with el Chavo         FAMILY     3.9     5933   
4500                Q*bert: Rebooted           GAME     4.3    13788   

                     Size  Installs  Price Content Rating         Android Ver  
10712  Varies with device   1000000    0.0       Everyone  Varies with device  
6167                  20M   1000000    0.0           Teen          2.3 and up  
3681                 4.1M   1000000    0.0       Everyone        4.0.3 and up  
9551                  60M   1000000    0.0       Everyone          4.1 and up  
4500                  55M   1000000    0.0       Everyone          4.4 and up

# Looking at top 5 apps from top 5 average most used genres

In [330]:
apple_most_popular_genres = apple_mask.groupby('prime_genre')['rating_count_tot'].mean().sort_values(ascending=False).head()
google_most_popular_genres = google_mask.groupby('Category')['Installs'].mean().round().sort_values(ascending=False).head()

In [332]:
for genre in apple_most_popular_genres.index:
    print(f"\nTop 5 apps in {genre} genre:")
    top_apps = apple_mask[apple_mask['prime_genre'] == genre]\
                 .sort_values(by='rating_count_tot', ascending=False)\
                 .head()
    display(top_apps)


Top 5 apps in Photo & Video genre:


track_name  size_bytes  price  \
1245  Baby Story - Pregnancy Pics Baby Milestones Photo   166354944   0.00   
1257  Capture - Control Your GoPro Camera - Share Video    94800896   0.00   
1265                                              Meitu   142101504   0.00   
1277                                           PicFrame    36758528   0.99   
1328     Visage makeup editor plus photo teeth whitener   134016000   0.00   

      rating_count_tot  user_rating cont_rating    prime_genre  \
1245              6700          4.5          4+  Photo & Video   
1257              6542          3.0         12+  Photo & Video   
1265              6478          5.0          4+  Photo & Video   
1277              6374          4.5          4+  Photo & Video   
1328              5767          4.0          4+  Photo & Video   

      sup_devices.num  lang.num  vpp_lic  
1245               37        12        1  
1257               37        10        1  
1265               37         5        1  
1277               37        21        1  
1328               37        10        1


Top 5 apps in Productivity genre:


track_name  size_bytes  price  \
1263                                    Verizon Cloud   104749056   0.00   
1281  Notepad+: Take Notes, Annotate and Write on PDF    63457280  14.99   
1287                                       Notes Plus   109620224   9.99   
1292                                     Duet Display    30325760   9.99   
1312             Home Design 3D - 3D Printing Edition   210538496   6.99   

      rating_count_tot  user_rating cont_rating   prime_genre  \
1263              6482          4.5          4+  Productivity   
1281              6288          4.5          4+  Productivity   
1287              6257          4.0          4+  Productivity   
1292              6165          4.5          4+  Productivity   
1312              6001          4.5          4+  Productivity   

      sup_devices.num  lang.num  vpp_lic  
1263               37         1        1  
1281               24        12        1  
1287               37        11        1  
1292               37        15        1  
1312               37        16        1


Top 5 apps in Business genre:


track_name  size_bytes  price  \
1286      InstaLogo Logo Creator - Graphic design maker   123656192   2.99   
1343  Tiny Scanner+ - PDF scanner to scan document, ...    57836544   4.99   
1679                                       HotSchedules    82037760   2.99   
1680                                        Uber Driver   162263040   0.00   
1709                  ScanBizCards Business Card Reader    96815104   0.99   

      rating_count_tot  user_rating cont_rating prime_genre  sup_devices.num  \
1286              6263          4.5          4+    Business               37   
1343              5611          5.0          4+    Business               38   
1679              3292          3.0          4+    Business               37   
1680              3289          3.0          4+    Business               37   
1709              3166          4.0          4+    Business               37   

      lang.num  vpp_lic  
1286        28        1  
1343         9        1  
1679         2        1  
1680        37        1  
1709         5        1


Top 5 apps in Health & Fitness genre:


track_name  size_bytes  price  \
1239        Seven - 7 Minute Workout Training Challenge   142803968   0.00   
1273      RUNNING for weight loss: workout & meal plans   110701568   0.00   
1324          Lifesum – Inspiring healthy lifestyle app   188017664   0.00   
1371  Windy ~ Sleep Relax Meditate & white noise sounds    95154176   1.99   
1409              Waterlogged - Daily Hydration Tracker    43439104   0.00   

      rating_count_tot  user_rating cont_rating       prime_genre  \
1239              6808          4.5          4+  Health & Fitness   
1273              6407          4.5          4+  Health & Fitness   
1324              5795          3.5          4+  Health & Fitness   
1371              5285          4.5          4+  Health & Fitness   
1409              5000          4.5         12+  Health & Fitness   

      sup_devices.num  lang.num  vpp_lic  
1239               37         8        1  
1273               37        10        1  
1324               37        11        1  
1371               37         2        1  
1409               37        10        1


Top 5 apps in Shopping genre:


track_name  size_bytes  price  \
1249    Flipp - Weekly Ads, Shopping List, and Coupons    92380160    0.0   
1272  AEO|Aerie: Jeans, Dresses, Swimsuits & Bralettes    64024576    0.0   
1357                                             Nike+   149407744    0.0   
1360                              Sam’s Club Scan & Go    32470016    0.0   
1406                 Close5 - Buy & Sell Stuff Locally    62879744    0.0   

      rating_count_tot  user_rating cont_rating prime_genre  sup_devices.num  \
1249              6677          4.5          4+    Shopping               37   
1272              6408          4.0          4+    Shopping               37   
1357              5464          5.0          4+    Shopping               37   
1360              5448          4.5          4+    Shopping               37   
1406              5010          4.0         17+    Shopping               37   

      lang.num  vpp_lic  
1249         2        1  
1272         1        1  
1357         1        1  
1360         1        1  
1406         1        1

In [334]:
for genre in google_most_popular_genres.index:
    print(f"\nTop 5 apps in {genre} genre:")
    top_apps = google_mask[google_mask['Category'] == genre]\
                 .sort_values(by='Installs', ascending=False)\
                 .head()
    display(top_apps)


Top 5 apps in ENTERTAINMENT genre:


App       Category  Rating  \
857                           Pluto TV - It’s Free TV  ENTERTAINMENT     4.2   
954                                  Showtime Anytime  ENTERTAINMENT     3.7   
882  🔥 Football Wallpapers 4K | Full HD Backgrounds 😍  ENTERTAINMENT     4.7   
952                                          LEGO® TV  ENTERTAINMENT     3.7   
918                                               VH1  ENTERTAINMENT     4.1   

     Reviews                Size  Installs  Price Content Rating  \
857    28948  Varies with device   1000000    0.0           Teen   
954    18523  Varies with device   1000000    0.0           Teen   
882    11661                4.0M   1000000    0.0       Everyone   
952    17247                7.2M   1000000    0.0   Everyone 10+   
918    27424                 17M   1000000    0.0           Teen   

            Android Ver  
857  Varies with device  
954  Varies with device  
882        4.0.3 and up  
952          5.0 and up  
918          4.4 and up


Top 5 apps in EDUCATION genre:


App   Category  Rating  Reviews  \
763           Programming Hub, Learn to code  EDUCATION     4.3    55704   
766  Socratic - Math Answers & Homework Help  EDUCATION     4.6    37862   
759                        Learn programming  EDUCATION     4.1    12733   
814               myHomework Student Planner  EDUCATION     4.0    28392   
750                                Brilliant  EDUCATION     4.5    41185   

                   Size  Installs  Price Content Rating         Android Ver  
763                 15M   1000000    0.0       Everyone          4.1 and up  
766                5.4M   1000000    0.0       Everyone          4.1 and up  
759  Varies with device   1000000    0.0       Everyone  Varies with device  
814  Varies with device   1000000    0.0       Everyone  Varies with device  
750  Varies with device   1000000    0.0       Everyone  Varies with device


Top 5 apps in WEATHER genre:


App Category  Rating  Reviews  Size  \
9491             Local weather Forecast  WEATHER     4.4     5482   12M   
3648   Klart.se - Sweden's best weather  WEATHER     3.6     7623   19M   
10148         EZ Clock & Weather Widget  WEATHER     4.2     6849  1.2M   
10058           Moonlight GO Weather EX  WEATHER     4.2    11510  3.5M   
3650                          Info BMKG  WEATHER     4.3    21404  7.6M   

       Installs  Price Content Rating   Android Ver  
9491    1000000    0.0       Everyone  4.0.3 and up  
3648    1000000    0.0       Everyone    4.4 and up  
10148   1000000    0.0       Everyone    2.1 and up  
10058   1000000    0.0       Everyone    2.2 and up  
3650    1000000    0.0       Everyone    4.4 and up


Top 5 apps in HOUSE_AND_HOME genre:


App        Category  \
1474  Home Security Camera WardenCam - reuse old phones  HOUSE_AND_HOME   
1463                                  Living Smart Home  HOUSE_AND_HOME   
1476                           Rent.com Apartment Homes  HOUSE_AND_HOME   
1462  Apartment, Home Rental Search: Realtor.com Ren...  HOUSE_AND_HOME   
1448                      Apartments & Rentals - Zillow  HOUSE_AND_HOME   

      Rating  Reviews                Size  Installs  Price Content Rating  \
1474     4.3    43800  Varies with device   1000000    0.0       Everyone   
1463     4.6    39189  Varies with device   1000000    0.0       Everyone   
1476     4.0     7508                5.1M   1000000    0.0       Everyone   
1462     3.9    10117                 13M   1000000    0.0       Everyone   
1448     4.2    27386  Varies with device   1000000    0.0       Everyone   

             Android Ver  
1474  Varies with device  
1463  Varies with device  
1476          5.0 and up  
1462        4.0.3 and up  
1448  Varies with device


Top 5 apps in PHOTOGRAPHY genre:


App     Category  Rating  \
2801                           TouchNote: Cards & Gifts  PHOTOGRAPHY     4.1   
6166                            Auto Background Changer  PHOTOGRAPHY     4.0   
6627                                          BQ Camera  PHOTOGRAPHY     3.6   
6193  Cut Out : Background Eraser and background cha...  PHOTOGRAPHY     4.1   
2842                  DSLR Camera Hd Ultra Professional  PHOTOGRAPHY     4.2   

      Reviews                Size  Installs  Price Content Rating  \
2801    19232                 28M   1000000    0.0       Everyone   
6166    35188                6.2M   1000000    0.0       Everyone   
6627     2526  Varies with device   1000000    0.0       Everyone   
6193     7118                9.2M   1000000    0.0       Everyone   
2842     4410                4.5M   1000000    0.0       Everyone   

             Android Ver  
2801          4.4 and up  
6166          3.2 and up  
6627  Varies with device  
6193          4.0 and up  
2842        4.0.3 and up

# Findings
The photography genre appears in both the apple and google most average used genres.\
Most of the most popular apps in the photography genre are free to use.\

# Conlusion
Our analysis shows that Photography is one of the most popular app genres across both the Apple App Store and Google Play Store.\
Since the market is already proven, we can consider creating a photography app with a unique selling point such as AI-powered photo enhancement, niche filters (e.g., vintage film, artistic styles), social media integration, etc.

We can monetise the app by operating a freemium structure, providing in-app purchases (premium filters, no-watermark exports), or creating a ad-free paid-version 